webrender/src/device/gfx.rs

```diff
@@ -3603,22 +3603,22 @@ impl<B: hal::Backend> Device<B> {
     pub fn blit_render_target(&mut self, src_rect: DeviceIntRect, dest_rect: DeviceIntRect) {
         debug_assert!(self.inside_frame);
 
-        let (src_img, src_layer) = if self.bound_read_fbo != DEFAULT_READ_FBO {
+        let (src_format, src_img, src_layer) = if self.bound_read_fbo != DEFAULT_READ_FBO {
             let fbo = &self.fbos[&self.bound_read_fbo];
             let img = &self.images[&fbo.texture];
             let layer = fbo.layer_index;
-            (&img.core, layer)
+            (img.format, &img.core, layer)
         } else {
-            (&self.frame_images[self.current_frame_id], 0)
+            (ImageFormat::BGRA8, &self.frame_images[self.current_frame_id], 0)
```

Comment:
do we always have the main FBO with BGRA8 format? Let's have it defined somewhere as a constant then. Otherwise, let's get the actual format from whatever created the swapchain.